house/house.go

```diff
@@ -0,0 +1,369 @@
+		if houseDevice.Id.Ip == incoming.Ip &&
+			houseDevice.Id.Mac == incoming.Mac ||
+			houseDevice.Id.Mac == incoming.Mac && incoming.Mac != "" ||
+			houseDevice.Id.Ip == incoming.Ip && incoming.Mac == "" ||
+			houseDevice.Id.Ip != incoming.Ip && incoming.Mac != "" && incoming.Mac == houseDevice.Id.Mac ||
+			houseDevice.Id.Ip == incoming.Ip && incoming.Ip != "" && incoming.Mac != houseDevice.Id.Mac {
+
+			if incoming.Ip != "" {
+				houseDevice.Id.Ip = incoming.Ip
+			} else if incoming.Mac != "" {
+				houseDevice.Id.Mac = incoming.Mac
+			}
+			newDevice = false
+			log.Println(houseDevice.Name)
+			timeAway := s.DeviceDetectState(*houseDevice)
+			if timeAway > *timeAwaySeconds && houseDevice.Person {
+				log.Println(fmt.Sprintf("Device: %s has returned after %d seconds", houseDevice.Name, timeAway))
+				if *debug {
+					log.Printf("Notification: %s, %s", houseDevice.Name, "is home")
+				} else {
+					err := notifications.SendNotification(houseDevice.Name, "has returned home.")
+					if err != nil {
+						return nil, err
+					}
+				}
+
+			}
+			houseDevice.Away = false
+			houseDevice.LastSeen = int64(time.Now().Unix())
+		}
+	}
+	if newDevice {
+		name := in.Ip
+		if in.Mac != "" {
+			name = in.Mac
+		}
+		newDevice := device{
+			Name:     strings.ReplaceAll(strings.ReplaceAll(name, ".", "_"), ":", "_"),
+			Id:       networkId{Ip: in.Ip, Mac: in.Mac},
+			Away:     false,
+			LastSeen: int64(time.Now().Unix()),
+			Person:   false,
+			Command:  "",
+		}
+
+		log.Println(fmt.Printf("New Device: %s", name))
+		if !*debug {
+			err := notifications.SendNotification(newDevice.Name, "New Device")
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		houseDevices = append(houseDevices, &newDevice)
+		err := writeConfig("config/devices.yaml")
+		if err != nil {
+			return nil, err
+		}
+		s.registerMetric(newDevice)
+	}
+	return &pb.Reply{Acknowledged: true}, nil
+}
+
+func (s *Server) httpHealthCheck(url string) bool {
+	req, _ := http.NewRequest("GET", url, nil)
+
+	req.Header.Add("cache-control", "no-cache")
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false
+	}
+	defer res.Body.Close()
+	return res.StatusCode == 200
+}
+
+func (s *Server) isDeviceOn(iot *device) (bool, error) {
+	lastSeen := s.DeviceDetectState(*iot)
+	if lastSeen > int64(syncStatusWithGA) {
+		state, err := s.callAssistant(iot.Command)
+		if err != nil {
+			return false, nil
+		}
+		iot.LastSeen = int64(time.Now().Unix())
+		iot.Away = *state == "off"
+	}
+
+	return !iot.Away, nil
+}
+
+func (s *Server) isHouseEmpty() bool {
+	houseEmpty := true
+	for _, device := range houseDevices {
+		if !device.Away && device.Person {
+			houseEmpty = false
+		}
+	}
+	return houseEmpty
+}
+
+func (s *Server) iotStatusManager() error {
+	for _, device := range iotDevices {
+		err := s.IotDeviceManager(device)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *Server) IotDeviceManager(iotDevice *device) error {
```

Comment:
warning: exported method Server.IotDeviceManager should have comment or be unexported (golint)